From: Shannon Nelson <shannon.nelson@intel.com>
Date: Sat, 21 Feb 2015 06:45:28 +0000
Subject: [PATCH 386/497] i40e: check pointers before use
Patch-mainline: v4.0-rc3
Git-commit: 65d13461d7c3fa822ff2000d2c0fe5f5b1943cb9
References: bsc#922844 FATE#318525

Make sure we don't try to dereference NULL pointers when returning values
from the AdminQ calls.

Change-ID: Ia6694f2f415d50acf0aba063c863568742799aff
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Tested-by: Jim Young <james.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/intel/i40e/i40e_common.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/intel/i40e/i40e_common.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_common.c
@@ -2847,7 +2847,7 @@ i40e_status i40e_aq_add_udp_tunnel(struc
 
 	status = i40e_asq_send_command(hw, &desc, NULL, 0, cmd_details);
 
-	if (!status)
+	if (!status && filter_index)
 		*filter_index = resp->index;
 
 	return status;
